### PR TITLE
fix(kubectl): kubectl events doesn't filter events by GroupVersion fo…

### PR DESCRIPTION
**Pull request by Ithrael**
…r resource with full name

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Fix kubectl events doesn't filter events by group and version for resource with full name

#### Which issue(s) this PR fixes:

Fixes #   https://github.com/kubernetes/kubectl/issues/1470

#### Does this PR introduce a user-facing change?

```release-note
Fix kubectl events doesn't filter events by GroupVersion for resource with full name.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Ithrael** [comment]
I'm sorry, I accidentally closed this PR (https://github.com/kubernetes/kubernetes/pull/120051)  and can't reopen it. I've opened a new PR and added E2E tests. The unit test for this case relies entirely on the API request, which has been mocked, so I didn't add the unit test. Please help review, thank you. @ardaguclu

File: test/cmd/events.sh [hunk 1]
```diff
@@ -42,6 +42,62 @@ run_kubectl_events_tests() {
...
+    "name": "cronjobs.example.com"
+  },
+  "spec": {
+    "group": "example.com",
+    "scope": "Namespaced",
+    "names": {
+      "plural": "cronjobs",
+      "singular": "cronjob",
+      "kind": "Cronjob",
+      "shortNames": ["mycj"]
```
Comment (ardaguclu) [inline test/cmd/events.sh:60]: I think, no need to define a shortName and instead we can delete it with it's full name.

File: test/cmd/events.sh [hunk 2]
```diff
@@ -61,6 +117,10 @@ run_kubectl_events_tests() {
     output_message=$(kubectl events -n test-events --for Cronjob.v1.batch/pi "${kube_flags[@]:?}" 2>&1)
     kube::test::if_has_string "${output_message}" "Warning" "InvalidSchedule" "Cronjob/pi"
 
+    # Post-Condition: events not returns event for fully qualified Cronjob.v1.example.com/pi when --for flag is used
+    output_message=$(kubectl events -n test-events --for Cronjob.v1.example.com/pi "${kube_flags[@]:?}" 2>&1)
+    kube::test::if_has_not_string "${output_message}" "Warning" "InvalidSchedule" "Cronjob/pi"
```
Comment (ardaguclu) [inline test/cmd/events.sh:122]: Did you verify that this test is failing on master(without the patch)?

**Review by ardaguclu — COMMENTED** [review]
/ok-to-test

File: test/cmd/events.sh (hunk 1 above)
Comment (Ithrael) [reply test/cmd/events.sh:60]: Yes, that makes sense. I added it for convenience when testing, but it's not necessary.

File: test/cmd/events.sh (hunk 2 above)
Comment (Ithrael) [reply test/cmd/events.sh:122]: I did some testing locally. Before I made the changes, running 'kubectl events -n test-events --for Cronjob.v1.example.com/pi` returned the events for Cronjob.v1.batch. After I made my changes, running `kubectl events -n test-events --for Cronjob.v1.example.com/pi` returned no events, which is the expected result.

**Comment by ardaguclu** [comment]
Thanks

/release-note-none
/triage accepted
/priority backlog
/approve
/lgtm

**Comment by sftim** [comment]
If an end user could notice the fix, we should add a changelog entry.

**Comment by Ithrael** [comment]
> If an end user could notice the fix, we should add a changelog entry.

I think this scenario is relatively niche, but if users had this scenario before, the return value of kubectl events filters might be corrected.

**Pull request merged**